crates/driver/src/infra/solver/dto/solution.rs

```diff
@@ -10,201 +10,212 @@ use {
+                                            .inputs
+                                            .into_iter()
+                                            .map(|input| eth::Asset {
+                                                amount: input.amount,
+                                                token: input.token.into(),
+                                            })
+                                            .collect(),
+                                        outputs: interaction
+                                            .outputs
+                                            .into_iter()
+                                            .map(|input| eth::Asset {
+                                                amount: input.amount,
+                                                token: input.token.into(),
+                                            })
+                                            .collect(),
+                                        internalize: interaction.internalize,
+                                    },
+                                ))
+                            }
+                            Interaction::Liquidity(interaction) => {
+                                let liquidity = liquidity
+                                    .iter()
+                                    .find(|liquidity| liquidity.id == interaction.id)
+                                    .ok_or(super::Error(
+                                        "invalid liquidity ID specified in interaction",
+                                    ))?
+                                    .to_owned();
+                                Ok(competition::solution::Interaction::Liquidity(
+                                    competition::solution::interaction::Liquidity {
+                                        liquidity,
+                                        input: eth::Asset {
+                                            amount: interaction.input_amount,
+                                            token: interaction.input_token.into(),
+                                        },
+                                        output: eth::Asset {
+                                            amount: interaction.output_amount,
+                                            token: interaction.output_token.into(),
+                                        },
+                                        internalize: interaction.internalize,
+                                    },
+                                ))
+                            }
+                        })
+                        .try_collect()?,
+                    weth,
+                    solver: solver.clone(),
                 })
-                .try_collect()?,
-            weth,
-            solver,
-        })
+            })
+            .collect()
     }
 }
 
+#[derive(Debug, Deserialize)]
+pub struct Solutions {
+    solutions: Vec<Solution>,
+}
```

Comment:
Are we OK with this format? I picked this format because I think it might be easier to change the API in the future if we return an object rather than returning an array directly. But if others disagree I'd be happy to change this to just a plain old JSON array.